test/e2e/lua/dynamic_configuration.go

```diff
@@ -189,6 +249,24 @@ func ensureRequest(f *framework.Framework, host string) {
 	Expect(resp.StatusCode).Should(Equal(http.StatusOK))
 }
 
+func ensureRequestWithStatus(f *framework.Framework, host string, statusCode int) {
```

Comment:
now that we have this helper we can refactor `ensureRequest` to use this function